CADETProcess/processModel/flowSheet.py

```diff
@@ -582,164 +588,83 @@ def get_flow_rates(self, state=None):
-            for origin, flow_rate in origin_flow_rates[unit.name].items():
-                flow_rates[unit.name].origins[origin] = np.array(flow_rate)
+            i_vec = np.zeros(dimension_length)
+            for unit_name in flow_rates:
+                i_vec[self.get_unit_index(self.units_dict[unit_name])] = flow_rates[unit_name][i]
 
-        for unit in self.units:
-            if not isinstance(unit, Inlet):
-                flow_rate_in = np.sum(
-                    list(flow_rates[unit.name].origins.values()), axis=0
-                )
-                flow_rates[unit.name].total_in = flow_rate_in
-            if not isinstance(unit, Outlet):
-                flow_rate_out = np.sum(
-                    list(flow_rates[unit.name].destinations.values()), axis=0
-                )
-                flow_rates[unit.name].total_out = flow_rate_out
+            lsg_list[i] = np.linalg.solve(qMatrix, i_vec)
 
-        return flow_rates
+        return_flow_rates = Dict()
 
-    def solve_flow_rates(self, inlet_flow_rates, output_states, coeff=0):
-        """Solve flow rates of system using sympy.
+        qMatrix_help = np.zeros((dimension_length, dimension_length))
 
-        Because a simple 'push' algorithm cannot be used when closed loops are
-        present in a FlowSheet (e.g. SMBs), sympy is used to set up and solve
-        the system of equations.
+        # qMatrix_help is generated to calucated the origin rate of cstr ( in q-Matrix if cstr it is supressed)
 
-        Parameters
-        ----------
-        inlet_flow_rates: dict
-            Flow rates of Inlet UnitOperations.
-        output_states: dict
-            Output states of all UnitOperations.
-        coeff: int
-            Polynomial coefficient of flow rates to be solved.
+        for unit in self.connections:
+            for origin in self.connections[unit]['origins']:
+                qMatrix_help[self.get_unit_index(unit), self.get_unit_index(origin)] \
+                    = output_states[origin][self.connections[origin].destinations.index(unit)]
 
-        Returns
-        -------
-        solution : dict
-            Solution of the flow rates in the system
-
-        Notes
-        -----
-            Since dynamic flow rates can be described as cubic polynomials, the
-            flow rates are solved individually for all coefficients.
-            To comply to the interface, the flow rates are always computed for the first
-            (constant) coefficient. For higher orders, the function returns None if all
-            coefficients are zero.
-
-            The problem could definitely be solved more elegantly (and efficiently) by
-            using proper liner algebra.
-        """
-        if len(inlet_flow_rates) == 0:
-            return None
+        #calculates the total in as matrix in "one" step rather then to iterate manually
+        total_in_matrix = qMatrix_help @ lsg_list.T
```

Comment:
Just out of curiosity, what's the `@` doing here?

Reply:
@ ist an Operator for matrix multiplikation. It's a little bit uneconomical because i calculate every time every input for every coefficent for every unit but i thought it was quite elegant.

Reply:
maybe i should rename lsg_list to lsg_matrix